massa-execution-worker/src/controller.rs

```diff
@@ -72,13 +73,23 @@ impl ExecutionController for ExecutionControllerImpl {
     /// * blockclique: new blockclique, replaces the current one in the input
     fn update_blockclique_status(
         &self,
-        finalized_blocks: HashMap<Slot, BlockId>,
-        new_blockclique: HashMap<Slot, BlockId>,
+        finalized_blocks: Map<BlockId, Slot>,
+        blockclique: Map<BlockId, Slot>,
     ) {
+        // index newly finalized blocks by slot
+        let mapped_finalized_blocks: HashMap<_, _> = finalized_blocks
+            .into_iter()
+            .map(|(b_id, slot)| (slot, b_id))
+            .collect();
+        // index blockclique by slot
+        let mapped_blockclique = blockclique
+            .into_iter()
+            .map(|(b_id, slot)| (slot, b_id))
+            .collect();
```

Comment:
So to me mapping the `Map<BlockId, Slot>` to `HashMap<Slot, BlockId>` here, instead of simply having the `HashMap<Slot, BlockId>` to begin with, seems to make things more complicated and I also don't see how swapping one for the other can have an effect on the logic. 

So I'd say let's stick to what makes sense, and try to diagnose the actual problem instead.